src/ray/common/task/task_spec.cc

```diff
@@ -215,6 +215,23 @@ ObjectID TaskSpecification::ReturnId(size_t return_index) const {
   return ObjectID::FromIndex(TaskId(), return_index + 1);
 }
 
+size_t TaskSpecification::NumStreamingGeneratorReturns() const {
+  return message_->num_streaming_generator_returns();
+}
+
+ObjectID TaskSpecification::StreamingGeneratorReturnId(size_t generator_index) const {
```

Comment:
Since the same index will generate the same ID now, we can infer every return IDs created from a task using the num_returns here.